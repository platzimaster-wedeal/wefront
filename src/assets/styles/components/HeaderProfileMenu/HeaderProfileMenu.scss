@import "../../partials/colors.scss";
@import "../../partials/fonts.scss";
@import "../../FontSize.scss";
@import "../../partials/text.scss";
@import "../../partials/sizes.scss";

.HeaderProfileMenu__container {
 background: #ffffff;
 box-shadow: 2px 3px 12px rgba(116, 116, 126, 0.3);
 border-radius: 30px;
 height: fit-content;
 max-height: 806px;
 width: fit-content;
 max-width: 679px;
 padding: 20px 26px 33px;
 position: relative;

 hr {
  color: $color-primary;
  border: 2px solid $color-primary;
  border-radius: 3px;
  margin-bottom: 24px;
 }
}

.HeaderProfileMenu--name {
 align-self: flex-end;
 height: fit-content;
 color: $color-primary;
 line-height: 40px;
 margin-right: 100px;

 span {
  font-family: $font-secondary;
  font-style: italic;
  font-weight: 200;
  color: $color-black-60;
  margin-top: -10px;
  @include font-size($font-size-normal + 8);
 }
}

.HeaderProfileMenu__name--profileImg {
 display: flex;
 justify-content: space-between;
 margin-bottom: 10px;
}

.HeaderProfileMenu--profileImg {
 width: 70px;
 height: 70px;
 object-fit: cover;
 border-radius: 50%;
 border: 3px solid $color-primary;
 cursor: pointer;
}

.HeaderProfileMenu__nav--ul {
 display: flex;
 flex-direction: column;
 justify-content: space-between;

 li {
  font-weight: 700;
  @include font-size($font-size-normal + 8);
  padding: 21px 0;
  border-top: 1px solid $color-black-30;
 }

 li:last-of-type {
  border-bottom: 1px solid $color-black-30;
 }
}
.nav__ul--link {
 color: #0a3525;
}

.nav__ul--link:hover,
.nav__ul--link:active {
 color: $color-primary;
}

.HeaderProfileMenu__logout {
 width: 100%;
 display: flex;
 align-items: center;
 justify-content: center;
 margin-top: 50px;
 margin-bottom: 30px;
 color: #0a3525;
 font-weight: 700;
 @include font-size($font-size-normal + 4);

 svg {
  width: 35px;
  height: auto;
  font-weight: 200;
  color: #b9b9bf;
  margin-right: 5px;
 }
}

.HeaderProfileMenu__footer {
 width: 100%;
 display: flex;
 flex-direction: column;
 align-items: center;
 justify-content: center;
}

.HeaderProfileMenu__footer--terms {
 font-style: italic;
 font-weight: 200;
 color: $color-black-60;
 margin-bottom: 20px;
 @include font-size($font-size-small + 1);

 a {
  color: $color-black-60;
 }

 a:hover {
  color: $color-primary;
 }
}

.HeaderProfileMenu__footer--logo {
 @include font-size($font-size-normal + 3);
 color: $color-primary;
 text-align: center;
 margin-top: 30px;

 @include media(tablet) {
  position: absolute;
  right: 26px;
  bottom: 65px;
  @include font-size($font-size-large);
 }
}
