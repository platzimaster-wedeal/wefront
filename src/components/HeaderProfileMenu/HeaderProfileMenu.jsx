import React, { useState } from "react";
import { Link } from "react-router-dom";
import { MdExitToApp } from "react-icons/md";
import "../../assets/styles/components/HeaderProfileMenu/HeaderProfileMenu.scss";

const HeaderProfileMenu = ({ profileName, profileImg, profileType }) => {
 const [displayNemu, setDisplayMenu] = useState(false);

 const handleDisplayOn = () => {
  setDisplayMenu(true);
 };

 const handleDisplayOff = () => {
  setDisplayMenu(false);
 };

 const ProfileImage = (
  <img
   className="HeaderProfileMenu--profileImg"
   src={profileImg}
   alt={profileImg}
   onClick={() => handleDisplayOn()}
  />
 );

 const ProfileMenuContent = (
  <div className="HeaderProfileMenu__container">
   <div className="HeaderProfileMenu__name--profileImg">
    <h2 className="HeaderProfileMenu--name">
     {profileName}
     <br />
     <span>{profileType}</span>
    </h2>
    <img
     className="HeaderProfileMenu--profileImg"
     src={profileImg}
     alt={profileImg}
     onClick={() => handleDisplayOff()}
    />
   </div>
   <hr />
   <nav className="HeaderProfileMenu__nav">
    <ul className="HeaderProfileMenu__nav--ul">
     <li>
      <Link className="nav__ul--link" to="/draw">
       Home
      </Link>
     </li>
     <li>
      <Link className="nav__ul--link" to="/draw">
       My Deals
      </Link>
     </li>
     <li>
      <Link className="nav__ul--link" to="/draw">
       Problems
      </Link>
     </li>
     <li>
      <Link className="nav__ul--link" to="/draw">
       Profile
      </Link>
     </li>
     <li>
      <Link className="nav__ul--link" to="/draw">
       Configutation
      </Link>
     </li>
    </ul>
   </nav>
   <div className="HeaderProfileMenu__logout">
    <MdExitToApp />
    <p>Logout</p>
   </div>
   <div className="HeaderProfileMenu__footer">
    <p className="HeaderProfileMenu__footer--terms">
     <Link to="/draw">Terms of service</Link>
    </p>
    <p className="HeaderProfileMenu__footer--terms">
     Â© Copyright 2020 - WeDeal{" "}
    </p>
   </div>
   <h1 className="HeaderProfileMenu__footer--logo">W</h1>
  </div>
 );

 return <div>{displayNemu ? ProfileMenuContent : ProfileImage}</div>;
};
export default HeaderProfileMenu;
