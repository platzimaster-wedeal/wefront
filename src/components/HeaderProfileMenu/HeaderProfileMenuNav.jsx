import React from "react";
import { Link } from "react-router-dom";
import { MdExitToApp } from "react-icons/md";

import "../../assets/styles/components/HeaderProfileMenu/HeaderProfileMenu.scss";

const HeaderProfileNav = ({ profileName, profileType }) => {
 return (
  <div>
   <div className="HeaderProfileMenu__name--profileImg">
    <h2 className="HeaderProfileMenu--name">
     {profileName}
     <br />
     <span>{profileType}</span>
    </h2>
   </div>
   <hr />
   <nav className="HeaderProfileMenu__nav">
    <ul className="HeaderProfileMenu__nav--ul">
     <li>
      <Link className="nav__ul--link" to="/home">
       Home
      </Link>
     </li>
     <li>
      <Link className="nav__ul--link" to="/user/deals">
       My Deals
      </Link>
     </li>
     <li>
      <Link className="nav__ul--link" to="/problems">
       Problems
      </Link>
     </li>
     <li>
      <Link className="nav__ul--link" to="/user/profile">
       Profile
      </Link>
     </li>
     <li>
      <Link className="nav__ul--link" to="/user/configuration">
       Configutation
      </Link>
     </li>
    </ul>
   </nav>
   <div className="HeaderProfileMenu__logout">
    <MdExitToApp />
    <p>Logout</p>
   </div>
   <div className="HeaderProfileMenu__footer">
    <p className="HeaderProfileMenu__footer--terms">
     <Link to="/draw">Terms of service</Link>
    </p>
    <p className="HeaderProfileMenu__footer--terms">
     Â© Copyright 2020 - WeDeal{" "}
    </p>
   </div>
  </div>
 );
};

export default HeaderProfileNav;
