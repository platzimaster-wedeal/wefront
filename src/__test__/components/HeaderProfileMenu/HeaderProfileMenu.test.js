import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import HeaderProfileMenu from "../../../components/HeaderProfileMenu/HeaderProfileMenu";

describe("<HeaderProfileMenu/>", () => {
 const headerProfileMenu = mount(<HeaderProfileMenu/>);
 test("Render del componente HeaderProfileMenu", () => {
  expect(headerProfileMenu.length).toEqual(1);
 });
});

describe("headerProfileMenu Snapshot", () => {
 test("Comprobar la UI del componente HeaderProfileMenu", () => {
  const headerProfileMenu = create(<HeaderProfileMenu />);
  expect(headerProfileMenu.toJSON()).toMatchSnapshot();
 });
});