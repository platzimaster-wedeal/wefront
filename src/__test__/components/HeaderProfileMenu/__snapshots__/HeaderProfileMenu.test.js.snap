// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headerProfileMenu Snapshot Comprobar la UI del componente HeaderProfileMenu 1`] = `
<div>
  <img
    className="HeaderProfileMenu--profileImg"
    onClick={[Function]}
  />
</div>
`;
